packages/tanstack-react-query/src/internals/subscriptionOptions.ts

```
@@ -0,0 +1,286 @@
+  const resultRef = React.useRef<$Result>(getInitialState());
+
+  const [state, setState] = React.useState<$Result>(
+    trackResult(resultRef.current, addTrackedProp),
+  );
+
+  state.reset = reset;
+
+  const updateState = React.useCallback(
+    (callback: (prevState: $Result) => $Result) => {
+      const prev = resultRef.current;
+      const next = (resultRef.current = callback(prev));
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)